woocommerce/woocommerce
# Add to Cart Form - Stepper Layout: Ensure Compliance with min, max, and step HTML Attributes

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR implements changes to the Add to Cart form’s stepper layout to ensure that the min, max, and step attributes are properly respected.


Closes #52489 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable TT4.
2. Edit the Single Product Template.
3. Ensure that you have the Add To Cart Form block added.
4. Click on it.
5. Open the block settings.
6. Ensure that you have the `Quantity Selector` option
7. Ensure that the default one is `Input`.
8. Ensure that the input looks like this image:

![image](https://github.com/woocommerce/woocommerce/assets/4463174/a2263956-e591-4dc4-ac63-33695cb6d6ee)


10. Switch the option to `Stepper`
11. Ensure that the input looks like this image:
![image](https://github.com/woocommerce/woocommerce/assets/4463174/49585883-1798-4353-afb4-62083512d278)
**The following steps should be checked for both options and with a Simple Product, a Variation Product, and a Grouped Product**
12. Install [Min/Max Quantities for WooCommerce](https://woocommerce.com/it/products/minmax-quantities/).
13. Edit a Product.
14. Configure the Min/Max Quantities:
![image](https://github.com/user-attachments/assets/69b20461-3cd3-42d6-bbbc-adea5cf6725e)

15. Visit the Product.
16. Ensure that the settings set earlier are respected.
17. Ensure that the input works correctly (pay attention to focus/border/etc...).
18. Click Add to Cart.
19. Ensure that the right amount of the products is added with the right price.



### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Add to Cart Form: Ensure compliance with min, max, and step HTML attributes.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
